### ISSUE TITLE
Problems Faced during Training Networks

### ISSUE THREAD

**Issue by shubhamgupta-dat** [shubhamgupta-dat/learning-cnn#1]
This will be a list of problems that developer usually face while developing NN architectures. I will try to list and mark the topics that I have covered in the documentation. 

You will see a red heart if it is covered. ❤️

**Comment by shubhamgupta-dat** [comment]
Class Imbalance dealt with:
- Weighted Loss Functions
- Resampling (Undersampling and Oversampling)

**Comment by shubhamgupta-dat** [comment]
Dataset Size Challenge
- Transfer Learning
- Data Augmentation

**Comment by shubhamgupta-dat** [comment]
Data Overlap Challenge

**Comment by shubhamgupta-dat** [comment]
Set Sampling Challenge
- Minimum percentage of Minority Class in Test Set like label is 15% but we sample it to be at least 30%. Similar can be done validation before we start training the remaining labels.

**Comment by shubhamgupta-dat** [comment]
Alignment Problem:
- Different Channels do overlap well with each other
- One way to solve this problem is **Image Registration**
- It is useful to be aware of image registration as a useful tool when trying to combine 3D volumes

**Comment by shubhamgupta-dat** [comment]
Overfitting
